src/emu/debug/debugcmd.cpp

```diff
@@ -517,70 +521,87 @@ bool debugger_commands::mini_printf(std::ostream &stream, std::string_view forma
+					else
 					{
 						m_console.printf("Not enough parameters for format!\n");
 						return false;
 					}
-					if (u32(*param >> 32) != 0)
-						util::stream_format(stream, zerofill ? "%0*X" : "%*X", (width <= 8) ? 1 : width - 8, u32(*param >> 32));
-					else if (width > 8)
-						util::stream_format(stream, zerofill ? "%0*X" : "%*X", width - 8, 0);
-					util::stream_format(stream, zerofill ? "%0*X" : "%*X", (width < 8) ? width : 8, u32(*param));
-					param++;
-					params--;
 					break;
 
 				case 'O':
 				case 'o':
-					if (params == 0)
+					if (param < params.size() && m_console.validate_number_parameter(params[param++], number))
+						util::stream_format(stream, zerofill ? "%0*o" : "%*o", width, number);
+					else
 					{
 						m_console.printf("Not enough parameters for format!\n");
 						return false;
 					}
-					if (u32(*param >> 60) != 0)
-					{
-						util::stream_format(stream, zerofill ? "%0*o" : "%*o", (width <= 20) ? 1 : width - 20, u32(*param >> 60));
-						util::stream_format(stream, "%0*o", 10, u32(BIT(*param, 30, 30)));
-					}
-					else
-					{
-						if (width > 20)
-							util::stream_format(stream, zerofill ? "%0*o" : "%*o", width - 20, 0);
-						if (u32(BIT(*param, 30, 30)) != 0)
-							util::stream_format(stream, zerofill ? "%0*o" : "%*o", (width <= 10) ? 1 : width - 10, u32(BIT(*param, 30, 30)));
-						else if (width > 10)
-							util::stream_format(stream, zerofill ? "%0*o" : "%*o", width - 10, 0);
-					}
-					util::stream_format(stream, zerofill ? "%0*o" : "%*o", (width < 10) ? width : 10, u32(BIT(*param, 0, 30)));
-					param++;
-					params--;
 					break;
 
 				case 'D':
 				case 'd':
-					if (params == 0)
+					if (param < params.size() && m_console.validate_number_parameter(params[param++], number))
+						util::stream_format(stream, zerofill ? "%0*d" : "%*d", width, number);
+					else
 					{
 						m_console.printf("Not enough parameters for format!\n");
 						return false;
 					}
-					util::stream_format(stream, zerofill ? "%0*d" : "%*d", width, u32(*param));
-					param++;
-					params--;
 					break;
+
 				case 'C':
 				case 'c':
-					if (params == 0)
+					if (param < params.size() && m_console.validate_number_parameter(params[param++], number))
+						stream << char(number);
+					else
 					{
 						m_console.printf("Not enough parameters for format!\n");
 						return false;
 					}
-					stream << char(*param);
-					param++;
-					params--;
 					break;
 
+				case 's':
+					if ((param + 1) < params.size())
+					{
+						address_space *space, *tspace;
+						std::string s;
+
+						if (m_console.validate_device_space_parameter(params[param + 0], -1, space) && m_console.validate_number_parameter(params[param + 1], number))
+						{
+							for (u32 address = u32(number), taddress; space->device().memory().translate(space->spacenum(), device_memory_interface::TR_READ, taddress = address, tspace); address++)
```

Comment:
`tspace` and `s` can be moved inside the `if`, can’t they?

Reply:
Yes, done.